File: primes.c
#include <stdio.h>
#include <stdbool.h>
#include <time.h>
#include <math.h>
#include <unistd.h>

bool isPrime(int number) {
    if (number <= 1) return false;
    for (int div = 2; div * div <= number; div++) {
        if (number % div == 0) return false;
    }
    return true;
}

int main() {
    printf("C program to get Prime numbers.\n");
    printf("Enter the range: >>>");
    int num_range;
    scanf("%d", &num_range);

    clock_t tm1, tm2;
    tm1 = clock();

    int i = 0;
    printf("\n");
    for (int num = 2; num <= num_range; num++) {
        if (isPrime(num)) {
            i++;
            printf("%*d --> prime   | %d\n", (int)(1 + log10(num_range)), num, i);
        }
    }

    tm2 = clock();

    printf("Time: %f seconds\n", (double)(tm2 - tm1) / CLOCKS_PER_SEC);
    printf("Quit after 60 seconds\n");
    sleep(60);

    return 0;
}

File: primes.py
from time import time as tm
from time import sleep as slp

def Is_prime(number):
    for div in range(2, number, 2):   # Chech if the number is divisible to all the number the less than it,
        if (number % div) == 0 :      # If it retutn 0 that mean the munber is divisible , So is not prime.
            return False
    return True                       # Other wise number is prime.


num_range = int(input("Python script to get a Prime numbers.\nEnter the range >>> "))
print()
tm1 = tm()
i = 0
for num in range((num_range - 1)):

    if Is_prime((num + 2)):
        i += 1
        znum = str((num + 2)).zfill(len(str(num_range)))
        print(f"{znum} --> prime   | {i}")

tm2 = tm()

print (f"time = {tm2 - tm1}")

print("Quit after 60 seconds")
slp(60)
